src/variables/tree.tsx

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { ArrayExt } from '@lumino/algorithm';
+
+import { convertType } from '.';
+
+import { ReactWidget } from '@jupyterlab/apputils';
+
+import React, { useEffect, useState } from 'react';
+
+import { IDebugger } from '../tokens';
+
+import { VariablesModel } from './model';
+
+/**
+ * The body for Tree Variables .
+ */
+export class VariablesBodyTree extends ReactWidget {
+  /**
+   * Instantiate a new Body for the Tree Variables.
+   * @param model The model for the variables.
+   */
+  constructor(options: VariablesBodyTree.IOptions) {
+    super();
+    this._model = options.model;
+    this._service = options.service;
+    this.addClass('jp-DebuggerVariables-body');
+    this._model.changed.connect(this.updateScopes, this);
+  }
+
+  private updateScopes(self: VariablesModel) {
+    if (ArrayExt.shallowEqual(this._scopes, self.scopes)) {
+      return;
+    }
+    this._scopes = self.scopes;
+    this.update();
+  }
+
+  /**
+   * Render the VariablesComponent.
+   */
+  render() {
+    return (
+      <>
+        {this._scopes.map(scope => (
+          <VariablesComponent
+            key={scope.name}
+            service={this._service}
+            data={scope.variables}
+          />
+        ))}
+      </>
+    );
+  }
+
+  private _model: VariablesModel;
+  private _scopes: VariablesModel.IScope[] = [];
+  private _service: IDebugger;
+}
+
+/**
+ * A React component to display a list of variables.
+ * @param model The model for the variables.
+ */
+const VariablesComponent = ({
+  data,
+  service
+}: {
+  data: VariablesModel.IVariable[];
+  service: IDebugger;
+}) => {
+  const [variables, setVariables] = useState(data);
+
+  useEffect(() => {
+    setVariables(data);
+  }, [data]);
+
+  return (
+    <>
+      <ul>
+        {variables.map(variable => {
+          return (
+            <VariableComponent
+              key={variable.evaluateName}
+              data={variable}
+              service={service}
+            />
+          );
+        })}
+      </ul>
+    </>
+  );
+};
+
+const VariableComponent = ({
+  data,
+  service
+}: {
+  data: VariablesModel.IVariable;
+  service: IDebugger;
+}) => {
+  const [variable] = useState(data);
+  const [expanded, setExpanded] = useState(null);
+  const [details, setDetails] = useState(null);
+  const stylesCaret = {
+    transform: `rotate(90deg)`
+  };
+
+  const styleName = {
+    color: 'var(--jp-mirror-editor-attribute-color)'
+  };
+
+  const styleType = {
+    color: 'var(--jp-mirror-editor-string-color)'
+  };
+
+  const nonExpanded =
```

Comment:
How about renaming this variable to something like `expandable`, and invert its value?